framework/src/kernels/NodalScalarKernel.C

```diff
@@ -12,27 +12,56 @@
 // MOOSE includes
 #include "Assembly.h"
 #include "MooseVariableScalar.h"
+#include "MooseMesh.h"
 #include "SystemBase.h"
 
 template <>
 InputParameters
 validParams<NodalScalarKernel>()
 {
   InputParameters params = validParams<ScalarKernel>();
-  params.addRequiredParam<std::vector<dof_id_type>>("nodes", "Node ids");
+  params.addParam<std::vector<dof_id_type>>("nodes", "Supply nodes using node ids");
+  params.addParam<std::vector<BoundaryName>>("nodeset_names", "Suppy nodes using named boundaries");
+
   return params;
 }
 
 NodalScalarKernel::NodalScalarKernel(const InputParameters & parameters)
   : ScalarKernel(parameters),
     Coupleable(this, true),
     MooseVariableDependencyInterface(),
-    _node_ids(getParam<std::vector<dof_id_type>>("nodes"))
+    _node_ids(getParam<std::vector<dof_id_type>>("nodes")),
+    _node_bc_names(getParam<std::vector<BoundaryName>>("nodeset_names"))
 {
   // Fill in the MooseVariable dependencies
   const std::vector<MooseVariable *> & coupled_vars = getCoupledMooseVars();
   for (const auto & var : coupled_vars)
     addMooseVariableDependency(var);
+
+  // Check if node_ids and/or node_bc_names given
+  if ((_node_ids.size() == 0) && (_node_bc_names.size() == 0))
+    mooseError("Must provide one of nodes or node_names");
+
+  if ((_node_ids.size() != 0) && (_node_bc_names.size() != 0))
+    mooseError("Must provide only one input using nodes or node_names");
+
+  // nodal bc names provided, append the nodes in each bc to _node_ids
+  if ((_node_ids.size() == 0) && (_node_bc_names.size() != 0))
+  {
+    std::vector<dof_id_type> nodelist;
+
+    for (auto i = beginIndex(_node_bc_names); i < _node_bc_names.size(); i++)
+    {
+      nodelist = _mesh.getNodeList(_mesh.getBoundaryID(_node_bc_names[i]));
+      for (auto k = beginIndex(_node_bc_names); k < nodelist.size(); k++)
+        _node_ids.push_back(nodelist[k]);
+    }
+  }
+  else
+  {
+    _console
+        << "Tip: NodalScalarKernel's can take in nodal bc names instead of the node numbers.\n";
```

Comment:
You want to use `mooseDeprecated` here, so people start seeing a warning that will force them to update their input files. We want to get rid of `nodes` eventually. It should say something like: "Using the 'node' parameter is deprecated. Please, update your input file to use the 'boundary' parameter."